src/components/RelatedAgreementsFieldArray/RelatedAgreementField.js

```diff
@@ -19,71 +20,86 @@ import { AppIcon, Pluggable } from '@folio/stripes/core';
 import { urls } from '../utilities';
 import css from '../styles.css';
 
-export default class RelatedAgreementField extends React.Component {
-  static propTypes = {
-    id: PropTypes.string,
-    input: PropTypes.shape({
-      name: PropTypes.string,
-      value: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
-    }).isRequired,
-    meta: PropTypes.shape({
-      error: PropTypes.node,
-      touched: PropTypes.bool,
-    }).isRequired,
-    onAgreementSelected: PropTypes.func.isRequired,
-    agreement: PropTypes.shape({
-      agreementStatus: PropTypes.shape({
-        label: PropTypes.string
-      }),
-      endDate: PropTypes.string,
-      id: PropTypes.string,
-      name: PropTypes.string,
-      label: PropTypes.string,
-      startDate: PropTypes.string,
+const propTypes = {
+  agreement: PropTypes.shape({
+    agreementStatus: PropTypes.shape({
+      label: PropTypes.string
     }),
-  }
+    endDate: PropTypes.string,
+    id: PropTypes.string,
+    name: PropTypes.string,
+    label: PropTypes.string,
+    startDate: PropTypes.string,
+  }),
+  id: PropTypes.string,
+  input: PropTypes.shape({
+    name: PropTypes.string,
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+  }).isRequired,
+  meta: PropTypes.shape({
+    error: PropTypes.node,
+    touched: PropTypes.bool,
+  }).isRequired,
+  onAgreementSelected: PropTypes.func.isRequired,
+  parentAgreementId: PropTypes.string,
+  parentAgreementName: PropTypes.string,
+};
+export default function RelatedAgreementField({
+  agreement,
+  id,
+  input,
+  meta,
+  onAgreementSelected,
+  parentAgreementId,
+  parentAgreementName,
+}) {
+  const [selfLinked, setSelfLinked] = useState(false);
```

Comment:
Can we please change this to `selfLinkedWarning` or something? It is supposed to stick around even after the field has been reset to undefined, and so is no longer selfLinked.